p2p/transport/websocket/websocket.go

```diff
@@ -187,7 +198,13 @@ func (t *WebsocketTransport) maDial(ctx context.Context, raddr ma.Multiaddr) (ma
 		return nil, err
 	}
 	isWss := wsurl.Scheme == "wss"
+
 	dialer := ws.Dialer{HandshakeTimeout: 30 * time.Second}
+	if t.UseReuseport() {
+		dialer.NetDial = func(network, address string) (net.Conn, error) {
+			return t.reuseTransport.DialContext(ctx, raddr)
+		}
+	}
 	if isWss {
 		sni := ""
 		sni, err = raddr.ValueForProtocol(ma.P_SNI)
```

Comment:
This will also need to have the reuse logic: https://github.com/libp2p/go-libp2p/blob/master/p2p/transport/websocket/websocket.go#L205